src/chocolatey.benchmark/helpers/ManagedProcessHelper.cs

```diff
@@ -0,0 +1,233 @@
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+            }
+
+            return nextProcess?.ProcessName;
+
+            //var processName = FindIndexedProcessName(process.Id);
+
+            //if (string.IsNullOrEmpty(processName))
+            //{
+            //    return null;
+            //}
+
+            //var parentProcess = FindPidFromIndexedProcessName(processName);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetParent(parentProcess);
+
+            //if (topLevelProcess == null)
+            //{
+            //    return parentProcess.ProcessName;
+            //}
+
+            //return topLevelProcess;
+        }
+
+        public static string GetParentFiltered(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+            Process selectedProcess = null;
+
+            while (true)
+            {
+                var processId = nextProcess?.Id ?? process.Id;
+
+                var processName = FindIndexedProcessName(processId);
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+
+                if (!IsIgnoredParent(nextProcess.ProcessName))
+                {
+                    selectedProcess = nextProcess;
+                }
+            }
+
+            return selectedProcess?.ProcessName;
+
+            //var processName = FindIndexedProcessName(process.Id);
+
+            //if (string.IsNullOrEmpty(processName))
+            //{
+            //    return null;
+            //}
+
+            //var parentProcess = FindPidFromIndexedProcessName(processName);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetParent(parentProcess);
+
+            //if (topLevelProcess == null || IsIgnoredParent(topLevelProcess))
+            //{
+            //    if (IsIgnoredParent(parentProcess.ProcessName))
+            //    {
+            //        return null;
+            //    }
+            //    else
+            //    {
+            //        return parentProcess.ProcessName;
+            //    }
+            //}
+
+            //return topLevelProcess;

```

Comment:
Are these comments needed?

Reply:
@AdmiringWorm is probably the person who should answer that, before I go removing them.

Reply:
This commented out code is more from a reference basis, ie the initial look into getting the process information. It can be safely removed as it is not needed anymore.